aptos-move/aptos-aggregator/src/delayed_field_extension.rs

```diff
@@ -351,177 +124,77 @@ impl Aggregator {
-    /// If transaction that is currently executing did not initialize it,
-    /// a new aggregator instance is created.
-    /// Note: when we say "aggregator instance" here we refer to Rust struct and
-    /// not to the Move aggregator.
-    pub fn get_aggregator(
-        &mut self,
-        id: AggregatorVersionedID,
-        max_value: u128,
-    ) -> PartialVMResult<&mut Aggregator> {
-        let aggregator = self
-            .aggregators
-            .entry(id.clone())
-            .or_insert_with(|| Aggregator {
-                id: id.clone(),
-                state: AggregatorState::Delta {
-                    speculative_start_value: SpeculativeStartValue::Unset,
-                    delta: SignedU128::Positive(0),
-                    history: DeltaHistory::new(),
-                },
-                max_value,
-            });
-        if aggregator.max_value != max_value {
-            Err(code_invariant_error(format!(
-                "Max value for the aggregator changed ({} -> {})",
-                aggregator.max_value, max_value
-            )))?;
-        }
-        Ok(aggregator)
-    }
-
-    /// Returns the number of aggregators that are used in the current transaction.
-    pub fn num_aggregators(&self) -> u128 {
-        self.aggregators.len() as u128
-    }
-
-    /// Creates and a new Aggregator with a given `id` and a `max_value`. The value
-    /// of a new aggregator is always known, therefore it is created in a data
-    /// state, with a zero-initialized value.
-    pub fn create_new_aggregator(&mut self, id: AggregatorVersionedID, max_value: u128) {
-        let aggregator = Aggregator {
-            id: id.clone(),
-            state: AggregatorState::Create { value: 0 },
-            max_value,
-        };
-        self.aggregators.insert(id.clone(), aggregator);
-        self.new_aggregators.insert(id);
-    }
-
-    /// If aggregator has been used in this transaction, it is removed. Otherwise,
-    /// it is marked for deletion.
-    pub fn remove_aggregator_v1(&mut self, id: AggregatorVersionedID) {
-        // Only V1 aggregators can be removed.
-        assert_matches!(id, AggregatorVersionedID::V1(_));
-
-        self.aggregators.remove(&id);
-
-        if self.new_aggregators.contains(&id) {
-            self.new_aggregators.remove(&id);
-        } else {
-            // This avoids cloning the state key.
-            let state_key = id.try_into().expect("V1 identifiers are state keys");
-            self.destroyed_aggregators.insert(state_key);
-        }
+    pub fn read_aggregator(
+        &self,
+        id: DelayedFieldID,
+        resolver: &dyn DelayedFieldResolver,
+    ) -> PartialVMResult<u128> {
+        Ok(self
+            .read_value(id, resolver, ReadPosition::AfterCurrentTxn)?
+            .into_aggregator_value()?)
     }
 
     pub fn snapshot(
         &mut self,
         aggregator_id: DelayedFieldID,
```

Comment:
remove aggregator_? or delayed

Reply:
this is actually an aggregator - snapshot can be called with aggregator as an argument